cmd/clusterctl/client/repository/components_test.go

```diff
@@ -584,16 +583,13 @@ func Test_fixRBAC(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := fixRBAC(tt.args.objs, tt.args.targetNamespace)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("got = %v, want %v", got, tt.want)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like in Test_componentsClient_Get